lua/presence/init.lua

```diff
@@ -613,8 +615,65 @@ function Presence.discord_event(on_ready)
     end
 end
 
+-- Check if the current project/parent is in blacklist
+function Presence:check_blacklist(buffer, parent_dirpath, project_dirpath)
+    local parent_dirname = nil
+    local project_dirname = nil
+
+    -- Parse parent/project directory name
+    if parent_dirpath then
+        parent_dirname = self.get_filename(parent_dirpath, self.os.path_separator)
+    end
+
+    if project_dirpath then
+        project_dirname = self.get_filename(project_dirpath, self.os.path_separator)
+    end
+
```

Comment:
Okay i see what you are trying to say, but even if we do not have `parent_dirpath` and `project_dirpath` we still need to loop over the `blacklist_table` to check if the current buffer is blacklisted or not.

Reply:
Ohh right right, that makes sense. In that case I'll resolve this!

This shouldn't really be a performance concern anyway since I assume users would only have a relatively small number of items in the blacklist. LGTM!